layer/migration.go

```diff
@@ -83,45 +83,45 @@ func (ls *layerStore) RegisterByGraphID(graphID string, parent ChainID, diffID D
 
 	if existingLayer := ls.get(layer.chainID); existingLayer != nil {
 		// Set error for cleanup, but do not return
-		err = errors.New("layer already exists")
+		cleanupErr = errors.New("layer already exists")
 		return existingLayer.getReference(), nil
 	}
 
-	tx, err := ls.store.StartTransaction()
-	if err != nil {
-		return nil, err
+	tx, cleanupErr := ls.store.StartTransaction()
```

Comment:
There's a few in here that use `:=` with the presumably-shared `cleanupErr` that _won't_ be using the shared variable, right? https://go.dev/play/p/8Weqd7BZJW6

Reply:
Trying to figure out which of the errors here (and below) should be considered a `cleanupErr` and which ones not. This function is much too involved 😞 

Perhaps I should revert these changes, then apply them again in a separate PR for closer review.